main.go

```diff
@@ -6,60 +6,126 @@ import (
 	"flag"
 	"fmt"
 	"io/fs"
+	"log"
 	"net"
 	"os"
 	"path/filepath"
 	"strconv"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	ItemTypeTextFile  = 0
+	ItemTypeDirectory = 1
+	// TODO: add more
 )
 
 func main() {
+	if err := do(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func do() error {
 	ex, err := os.Executable()
 	if err != nil {
-		panic(err)
+		return errors.Wrap(err, "getting executable")
```

Comment:
this is a convention from real world, but i use in side projects now because of how useful it is.

Reply:
(wrap all errs, always lower case always verb first, never say error/failed/etc)

Reply:
I like this